Applications/Spire/Source/OrderImbalanceIndicator/LocalOrderImbalanceIndicatorModel.cpp

```diff
@@ -0,0 +1,48 @@
+#include "Spire/OrderImbalanceIndicator/LocalOrderImbalanceIndicatorModel.hpp"
+
+using namespace boost::icl;
+using namespace Nexus;
+using namespace Spire;
+
+void LocalOrderImbalanceIndicatorModel::publish(
+    const Nexus::OrderImbalance& imbalance) {
+  m_publish_queue.push(imbalance);
+  if(m_publish_queue.size() > 1) {
+    return;
+  }
+  while(!m_publish_queue.empty()) {
+    const auto& current_imbalance = m_publish_queue.front();
+    if(!m_imbalances.insert(
+        {current_imbalance.m_security, current_imbalance}).second) {
+      if(auto& previous_imbalance = m_imbalances.at(imbalance.m_security);
+          previous_imbalance.m_timestamp < imbalance.m_timestamp) {
+        previous_imbalance = current_imbalance;
+      }
+    }
+    for(auto& subscription : m_subscriptions) {
+      if(contains(subscription.m_interval, current_imbalance.m_timestamp)) {
+        subscription.m_signal(current_imbalance);
+      }
+    }
+    m_publish_queue.pop();
+  }
+}
+
+SubscriptionResult<std::vector<Nexus::OrderImbalance>>
+    LocalOrderImbalanceIndicatorModel::subscribe(const TimeInterval& interval,
+      const OrderImbalanceSignal::slot_type& slot) {
+  m_subscriptions.push_back({interval});
```

Comment:
If subscribe is constantly called, the size of the  m_subscriptions will continue to increase. How to unsubscribe?

Reply:
The caller of subscribe can unsubscribe by disconnecting the connection returned to it in the SubscriptionResult. But, the Local model will have to be updated to clean up expired subscriptions.